doc/developers/contributing.rst

```diff
@@ -751,8 +751,11 @@ In general have the following in mind:
 * When editing reStructuredText (``.rst``) files, try to keep line length under
   80 characters when possible (exceptions include links and tables).
 
-* Before submitting you pull request check if your modifications have introduced
-  new sphinx warnings and try to fix them.
+* Do not modify label for internal links as this would break existing cross
```

Comment:
"labels" or "any label"

Also I think we need to clarify that the issue is with *URLs* (from external websites), which we have no control over. For internal cross-refs, we could just change the `:ref:` stuff.

Reply:
Maybe we can explain that labels used for internal links (with the `:ref:` directive) can also appear in the URL pointing to this section from an external website, and therefore should never be changed to avoid breaking incoming links to the scikit-learn documentation.